pkg/operator/ceph/cluster/controller_test.go

```diff
@@ -27,116 +29,255 @@ import (
+		// Make sure it has the fake CephCluster that is to be deleted in it
+		client := clientfake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(fakeCluster.DeepCopy()).Build()
+
+		// Create a ReconcileCephClient object with the scheme and fake client.
+		reconcileCephCluster := &ReconcileCephCluster{
+			client:            client,
+			scheme:            scheme,
+			context:           clusterdCtx,
+			clusterController: controller,
+		}
+
+		req := reconcile.Request{NamespacedName: nsName}
+
+		_, err := reconcileCephCluster.Reconcile(ctx, req)
+		assert.Error(t, err)
+		event := <-fakeRecorder.Events
+		assert.Contains(t, event, "CephBlockPools")
+		assert.Contains(t, event, "my-block-pool")
+
+		updatedCephCluster := &cephv1.CephCluster{}
+		err = client.Get(ctx, nsName, updatedCephCluster)
+		assert.NoError(t, err)
+		status := updatedCephCluster.Status
+		assert.Equal(t, cephv1.ConditionDeleting, status.Phase)
+		assert.Equal(t, cephv1.ClusterState(cephv1.ConditionDeleting), status.State)
+		assert.Equal(t, corev1.ConditionTrue, cephv1.FindStatusCondition(status.Conditions, cephv1.ConditionDeleting).Status)
+		assert.Equal(t, corev1.ConditionTrue, cephv1.FindStatusCondition(status.Conditions, cephv1.ConditionDeletionIsBlocked).Status)
+
+		t.Run("and deletion continues when dependents no longer exist", func(t *testing.T) {
+			gvr := cephv1.SchemeGroupVersion.WithResource("cephblockpools")
+			err = clusterdCtx.DynamicClientset.Resource(gvr).Namespace(cephNs).Delete(ctx, "my-block-pool", metav1.DeleteOptions{})
+			assert.NoError(t, err)
+
+			res, err := reconcileCephCluster.Reconcile(ctx, req)
+			assert.NoError(t, err)
+			assert.True(t, res.IsZero())
+			event := <-fakeRecorder.Events
+			assert.Contains(t, event, "ClusterDeleting")
 
-	// The test does not return volumes on the second call
-	assert.NoError(t, controller.checkIfVolumesExist(clusterToDelete))
+			updatedCephCluster := &cephv1.CephCluster{}
+			err = client.Get(ctx, nsName, updatedCephCluster)
+			status := updatedCephCluster.Status
+			assert.NoError(t, err)
+			assert.Equal(t, cephv1.ConditionDeleting, status.Phase)
+			assert.Equal(t, cephv1.ClusterState(cephv1.ConditionDeleting), status.State)
+			assert.Equal(t, corev1.ConditionTrue, cephv1.FindStatusCondition(status.Conditions, cephv1.ConditionDeleting).Status)
+			t.Log(cephv1.FindStatusCondition(status.Conditions, cephv1.ConditionDeletionIsBlocked))
+			assert.Equal(t, corev1.ConditionFalse, cephv1.FindStatusCondition(status.Conditions, cephv1.ConditionDeletionIsBlocked).Status)
+		})
+	})
+
+	t.Run("deletion blocked if PVs exist", func(t *testing.T) {
+		t.Run("but deletion continues if cleanupPolicy.allowUninstallWithVolumes=true", func(t *testing.T) {
+			t.Error("TODO: these tests")
+		})
+	})
 }
 
-func TestClusterDeleteFlexDisabled(t *testing.T) {
-	clusterName := "cluster684"
-	rookSystemNamespace := "rook-system-6413"
+func Test_checkIfVolumesExist(t *testing.T) {
```

Comment:
Renamed existing `TestClusterDelete` methods to be subtests under this one (`Test_checkIfVolumesExist`) since they weren't actually testing the full delete functionality and I needed to add full delete tests here.